File: app/src/main/java/com/sjani/usnationalparkguide/Data/CampDao.java
package com.sjani.usnationalparkguide.Data;


import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface CampDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void save(CampEntity campEntity);

    @Query("SELECT * FROM camps")
    LiveData<List<CampEntity>> getAllCamps();

    @Query("SELECT * FROM camps WHERE camp_id = :campId")
    LiveData<CampEntity> getCamp(String campId);

    @Query("DELETE FROM camps")
    void clearTable();

}

File: app/src/main/java/com/sjani/usnationalparkguide/data/dao/CampDao.kt
package com.sjani.usnationalparkguide.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.sjani.usnationalparkguide.data.entity.CampEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface CampDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun save(campEntity: CampEntity)
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun saveAll(camps: List<CampEntity>)
    
    @Query("SELECT * FROM camps")
    fun getAllCamps(): Flow<List<CampEntity>>
    
    @Query("SELECT * FROM camps WHERE camp_id = :campId")
    fun getCamp(campId: String): Flow<CampEntity?>
    
    @Query("DELETE FROM camps")
    suspend fun clearTable()
}


